app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(peoples: list) -> list:
+    list_result = []
+    new_person = None
+    [Person(member["name"], member["age"]) for member in peoples]
+
+    for member in peoples:
+        if "wife" in member and member["wife"]:
+            new_person = Person.people.get(member["name"])
+            new_person.wife = Person.people.get(member["wife"])
+            list_result.append(new_person)
```

Comment:
Use get and walrus operator to perform this operation efficiently. Apply this change for `husband` as well
```suggestion
        if wife_name := member.get("wife"):
            new_person = Person.people.get(member["name"])
            new_person.wife = Person.people.get(wife_name)
```
Besides that, retrieve `new_person` at the beginning of the loop to avoid the code duplication
```suggestion
    new_person = Person.people.get(member["name"])
        if wife_name := member.get("wife"):
            new_person.wife = Person.people.get(wife_name)
```